definitions/procedures/foreman_tasks/cleanup.rb

```diff
@@ -0,0 +1,53 @@
+module Procedures::ForemanTasks
+  class Cleanup < ForemanMaintain::Procedure
+    PERMITTED_STATES = %w[all pending scheduled planning planned running paused stopped].freeze
+    metadata do
+      for_feature :foreman_tasks
+      description 'Perform task cleanup'
+      preparation_steps { Checks::Foreman::DBUp.new }
+      confine do
+        feature(:foreman) && feature(:foreman_tasks)
+      end
+
+      param :states,
+            'Operate on tasks in STATES' \
+             "\nAvailable states: " + PERMITTED_STATES.join(', '),
+            :array => true, :allowed_values => PERMITTED_STATES
+      param :batch_size,
+            'Process tasks in batches of BATCH_SIZE, 1000 by default' do |s|
+        Integer(s) unless s.nil?
+      end
+      param :after,
+            'Operate on tasks older than AFTER. Expected format is a number ' \
+            "followed by the time unit (s,h,d,m,y), such as '10d' for 10 days"
+      param :backup, 'Backup deleted tasks', :flag => true
+      param :noop, 'Do a dry run, print what would be done', :flag => true
+      param :search, 'Use QUERY in scoped search format to match tasks to delete'
+      param :rake_command, 'Use RAKE_COMMAND as rake', :default => 'foreman-rake'
+      param :verbose, 'Be verbose', :flag => true
+      param :generate, 'Only generate the resulting command, do not run it', :flag => true
+    end
+
+    def run
+      f = feature(:foreman_tasks)
+
+      # If states are provided, then a search must be provided as well
+      # We assume if the user didn't provide a search query, they want to match
+      # all tasks in given stat so we provide a bogus search value which matches
+      # all tasks
+      @search = 'label != a_label' if !@states.empty? && @search.nil?
+      args = [@rake_command, @batch_size, @states, @after, @search, @backup, @noop, @verbose]
+
+      cleanup_command = f.generate_task_cleanup_command(*args)
+      if @generate
+        puts cleanup_command
+      else
+        message = 'Performing task cleanup'
+        with_spinner(message) do |spinner|
+          execute!(cleanup_command) { |line| spinner.update line }
```

Comment:
I can see we are passing the block,
~~~
================================================================================
Perform task cleanup: 
\ Performing task cleanup                                             [OK]       "2021-04-24 11:18:56") AND (state ^ (stopped))
--------------------------------------------------------------------------------
~~~

This is not the problem of your changes but the `@max_length` of CLIReporter is static. I have faced this issue few times. I have below options as proposal to handle this, let me know your thoughts,

* Use Highline.output_cols as `@max_length`. This will be dynamic value and will be set as per the terminal size, however this changes all the output. Which means we stretch the decorators `=`, `-'  and the status output too,

    With `Highline.output_cols`
    ~~~
    Running preparation steps required to run the next scenarios
    =============================================================================================================================================================================
    Make sure Foreman DB is up: 
    / Checking connection to the Foreman DB                                                                                                                            [OK]      
    ~~~

    Existing output,
    ~~~
    ================================================================================
    Make sure Foreman DB is up: 
    / Checking connection to the Foreman DB                               [OK]                                                                                                   
    ~~~

* The other way is we flush the line conditionally, I think this is better because first one will require to change the lots of messages which were written to fit in 80 charsize?
    ~~~
    class CLIReporter < Reporter
    ..
    ..
          def clear_line
            flush_size = @last_line.length > @max_length ? @last_line.length : @max_length
            print "\r" + ' ' * flush_size + "\r"
          end
    ~~~

Reply:
If I understand the issue correctly:
- When the task export is running, we are displaying its output line by line on the line with the spinner
- The lines are often too long leading to broken display

Is this correct?

Reply:
Yes!

Reply:
Sorry for the delay, done